### DISCUSSION TITLE
Feature: S3 Backup and other production features

### DISCUSSION THREAD

**Discussion by ifeulner** [mysticaltech/terraform-hcloud-kube-hetzner#368] [Ideas]
Hi, 

is there already a way to support additional production-needed feature like

- [x] adding S3 backup functionality at initial creation of the cluster
- [x] Enabling cluster secret encryption at Rest
- [ ] passing the CIS benchmark

Or did you guys discuss these topics already in the past? @mysticaltech? 

Thanks, Ingo.

**Comment by mysticaltech** [comment]
@ifeulner Good points above!

**Reply by ifeulner** [reply]
For a CIS-compliant install I find this blog interesting and helpful:  https://digitalis.io/blog/kubernetes/k3s-lightweight-kubernetes-made-ready-for-production-part-1/

**Comment by otavio** [comment]
The S3 backup, how are you guys intending to do it?

**Reply by ifeulner** [reply]
In the initial creation of the cluster. k3s supports direct backup/snapshotting of the cluster-state  to an S3 storage, see [here](https://docs.k3s.io/backup-restore#s3-compatible-api-support) for more details

Something like that:
```
k3s server \
  --cluster-init \
  --etcd-s3 \
  --etcd-s3-bucket=<S3-BUCKET-NAME> \
  --etcd-s3-access-key=<S3-ACCESS-KEY> \
  --etcd-s3-secret-key=<S3-SECRET-KEY>
```

Should not be too complicated to add this...
